cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -5102,6 +5102,10 @@ function(build_awssdk)
       find_program(PATCH patch REQUIRED)
       set(${BASE_VARIABLE_NAME}_PATCH_COMMAND
           ${PATCH} -p1 -i ${CMAKE_CURRENT_LIST_DIR}/aws-c-common-1208.patch)
+    elseif(MINGW AND AWSSDK_PRODUCT STREQUAL "aws-sdk-cpp")
+      find_program(PATCH patch REQUIRED)
+      set(${BASE_VARIABLE_NAME}_PATCH_COMMAND
+          ${PATCH} -p1 -i ${CMAKE_CURRENT_LIST_DIR}/aws-sdk-cpp.diff)
```

Comment:
Can we keep naming all patches with the `.patch` extension?

Reply:
(I'll remove this diff because this diff doesn't solve anything.)

In general, I'm using `.diff` for `diff -u`/`git diff` output and `.patch` for `git format-patch` output like GitHub does:
* https://github.com/apache/arrow/pull/46870.diff
* https://github.com/apache/arrow/pull/46870.patch

I used `.diff` here because it's just for testing. I'm using `.patch` for the final result.

Sorry for not marking this PR as draft.